vck-initializer/vck-initializer/main.go

```diff
@@ -0,0 +1,256 @@
+	if deployment.ObjectMeta.GetInitializers() != nil {
+		pendingInitializers := deployment.ObjectMeta.GetInitializers().Pending
+
+		if initializerName == pendingInitializers[0].Name {
+			log.Printf("Initializing deployment: %s", deployment.Name)
+
+			o := deployment.DeepCopyObject()
+			initializedDeployment := o.(*v1beta1.Deployment)
+
+			// Remove self from the list of pending Initializers while preserving ordering.
+			if len(pendingInitializers) == 1 {
+				initializedDeployment.ObjectMeta.Initializers = nil
+			} else {
+				initializedDeployment.ObjectMeta.Initializers.Pending = append(pendingInitializers[:0], pendingInitializers[1:]...)
+			}
+			if requireAnnotation {
+				a := deployment.ObjectMeta.GetAnnotations()
+				_, ok := a[annotation]
+				if !ok {
+					log.Printf("Required '%s' annotation missing; skipping vck initializing", annotation)
+					_, err := clientset.AppsV1beta1().Deployments(deployment.Namespace).Update(initializedDeployment)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
+				//log.Print("annotation: ", a[annotation])
+				info := &data{}
+				err := json.Unmarshal([]byte(a[annotation]), info)
+				if err != nil {
+					return err
+				}
+				fmt.Println("Unmarshal:", info.MountPath)
+				vckVM, err := crdClient.VckV1().VolumeManagers(deployment.GetNamespace()).Get(info.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				volumeVCK, affinityVCK, err := addVolumesAffinity(vckVM, info)
+				if err != nil {
+					return err
+				}
+				if info.Containers == nil {
+					for _, container := range deployment.Spec.Template.Spec.Containers {
+						info.Containers = append(info.Containers, container.Name)
+					}
+				}
+
+				for _, container := range info.Containers {
+					volumeMount, containerID, err := addVolumeMount(deployment, vckVM, container, info.MountPath)
+					if err != nil {
+						return err
+					}
+					initializedDeployment.Spec.Template.Spec.Containers[containerID].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[containerID].VolumeMounts, *volumeMount)
+				}
+				initializedDeployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, *volumeVCK)
+				initializedDeployment.Spec.Template.Spec.Affinity = affinityVCK
+
+			}
+			oldData, err := json.Marshal(deployment)
+			if err != nil {
+				return err
+			}
+
+			newData, err := json.Marshal(initializedDeployment)
+			if err != nil {
+				return err
+			}
+
+			patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1beta1.Deployment{})
+			if err != nil {
+				return err
+			}
+
+			_, err = clientset.AppsV1beta1().Deployments(deployment.Namespace).Patch(deployment.Name, types.StrategicMergePatchType, patchBytes)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func addVolumesAffinity(vckVM *vckv1.VolumeManager, info *data) (*corev1.Volume, *corev1.Affinity, error) {
```

Comment:
Do you want to maybe rename this to `getVolumesAffinity`?

Reply:
won't get just get the information but we are adding the object to the deployment?
not sure still.